blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -473,7 +485,15 @@ private MetricData transformColumnsToMetricData(Locator locator, ColumnList<Long
                                                                        Granularity gran) {
         try {
             RollupType rollupType = RollupType.fromString(metaCache.get(locator, rollupTypeCacheKey));
-            DataType dataType = new DataType(metaCache.get(locator, dataTypeCacheKey));
+            DataType dataType = getDataType(locator, dataTypeCacheKey);
+            if (rollupType == null) {
+                rollupType = RollupType.BF_BASIC;
+            }
+
+            if (dataType == null) {
+                dataType = DataType.NUMERIC;
+            }
+
```

Comment:
Fixed in  348c662
Turns out RollupType.fromString already had that support. Baked it in getDataType
However, in other places they are passed as params into public and protected methods. We need to be sure there that no one is passing nulls.